### ISSUE TITLE
Second use of PEP 695 generic function causes type error

### ISSUE THREAD

**Issue by jatoben** [microsoft/pyright#6386]
**Describe the bug**

When a generic function is defined with PEP 695 generic syntax, multiple calls in the same statement cause Pyright to raise an incompatible type error.

**Code or Screenshots**

```python
from collections.abc import Iterable

def only[T](items: Iterable[T]) -> T:
    ...

def foo(item_a: int, item_b: int) -> bool:
    ...

a = [1]

foo(only(a), only(a))
```

```
Argument of type "list[int]" cannot be assigned to parameter "items" of type "Iterable[type[T(1)@only]]" in function "only"
  "list[int]" is incompatible with "Iterable[type[T(1)@only]]"
    Type parameter "_T_co@Iterable" is covariant, but "int" is not a subtype of "type[T(1)@only]"
      Type "int" cannot be assigned to type "type[T(1)@only]"  (reportGeneralTypeIssues)
```

However, defining an equivalent function using `TypeVar` type-checks cleanly:

```python
from collections.abc import Iterable
from typing import TypeVar

_TV = TypeVar("_TV")
def only_tv(Items: Iterable[_TV]) -> _TV:
    ...

def foo(item_a: int, item_b: int) -> bool:
    ...

a = [1]

foo(only_tv(a), only_tv(a))
```

Looking at the type hints for each function, it appears that the PEP 695 version has an extra `type[]` indicator when disambiguating the type placeholder, whereas the `TypeVar`-using function does not:

```
(function) def only(items: Iterable[T@only]) -> T@only
(function) def only(items: Iterable[type[T(1)@only]]) -> type[T(1)@only]

(function) def only_tv(Items: Iterable[_TV@only_tv]) -> _TV@only_tv
(function) def only_tv(Items: Iterable[_TV(1)@only_tv]) -> _TV(1)@only_tv
```

[Playground link](https://pyright-play.net/?pyrightVersion=1.1.335&pythonVersion=3.12&strict=true&code=GYJw9gtgBAxmA28CmMAuBLMA7AzgOgEMAjGKdCABzBFSgElUkRjkAoUSKVATwvSwDmZStVoAVXkgBqBEK1YATJMCjZ43ANpiAugAp0jCDgBc9RsyLIt2gJRQAtAD4oY46ygeoeb-KUrgYGD6hgD6BKb8qAA0ZKFEEViodk5QRIHwbp5ePlmsBFAAvFAaAIza8gFBaty6BDYx1bU2NvIhYlKFLpIyILoARG1SfS1%2BqljqIagAbroMSEamcxZWg7YOzoOZnt54FYG61ZMzdQ3j3EdNLUA)

**VS Code extension or command-line**

Pyright command-line and as a language server, version 1.1.335

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

It's great to see that you're starting to use PEP 695 syntax!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.336, which I just published. It will also be included in a future release of pylance.
